kafka-client/src/main/java/org/apache/kafka/streams/state/internals/DelayedAsyncStoreBuilder.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * This source code is licensed under the Responsive Business Source License Agreement v1.0
+ * available at:
+ *
+ * https://www.responsive.dev/legal/responsive-bsl-10
+ *
+ * This software requires a valid Commercial License Key for production use. Trial and commercial
+ * licenses can be obtained at https://www.responsive.dev
+ */
+
+package org.apache.kafka.streams.state.internals;
+
+import dev.responsive.kafka.api.async.internals.stores.AbstractAsyncStoreBuilder;
+import java.lang.reflect.Field;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.internals.StoreFactory.FactoryWrappingStoreBuilder;
+import org.apache.kafka.streams.state.KeyValueBytesStoreSupplier;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.TimestampedBytesStore;
+
+public class DelayedAsyncStoreBuilder<K, V, T extends StateStore>
```

Comment:
this class is basically where all the nasty hacky stuff is confined to. For now there are two main culprits:

1. checking and unwrapping the `FactoryWrappingStoreBuilder`, an internal AK class I had to add while converting DSL operators to use #stores, so that the DSLStoreSuppliers are resolved correctly
2. some "light" reflection to grab the storeSupplier from a storeBuilder instance